crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -127,14 +136,19 @@ impl CoinGecko {
     async fn bulk_fetch_denominated_in_token(
         &self,
         mut tokens: HashSet<Token>,
-        denominator: Token,
     ) -> Result<HashMap<H160, NativePriceEstimateResult>, PriceEstimationError> {
-        tokens.insert(denominator);
-        let prices_in_eth = self.bulk_fetch_denominated_in_eth(&tokens).await?;
+        tokens.insert(self.denominator.address);
```

Comment:
Are we deliberately removing the optimisation of not requesting the price for weth if not needed?

Reply:
Yes, this was a conscious decision for the sake of simplicity. I'm happy to put the optimization back in if we actually need the extra 5% of batch size but for now I would prefer to keep the code leaner since this file is already on the more complex side IMO.